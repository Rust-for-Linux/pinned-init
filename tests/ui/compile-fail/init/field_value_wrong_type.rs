use pin_init::*;

struct Foo {
    a: usize,
}

fn main() {
    let _ = init!(Foo { a: () });
}
