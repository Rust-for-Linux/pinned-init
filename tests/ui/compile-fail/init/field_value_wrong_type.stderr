error[E0308]: mismatched types
 --> tests/ui/compile-fail/init/field_value_wrong_type.rs:8:28
  |
8 |     let _ = init!(Foo { a: () });
  |                         -  ^^ expected `usize`, found `()`
  |                         |
  |                         arguments to this function are incorrect
  |
note: function defined here
 --> $RUST/core/src/ptr/mod.rs
  |
  | pub const unsafe fn write<T>(dst: *mut T, src: T) {
  |                     ^^^^^
